### PR TITLE
feat(md): sort environments based on constraints on the main page

### PR DESCRIPTION
**Pull request by ranihorev**
Added a bunch of tests.

### CODE REVIEW

File: app/scripts/modules/core/src/managed/utils/sortEnvironments.spec.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            lifecycleSteps: [],
+          },
+        ],
+      },
+    ],
+  };
+  return { response, versionEnviroments };
+};
+
+fdescribe('sort environments', () => {
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/managed/utils/sortEnvironments.spec.ts:81]: `fdescribe` -> `describe`

File: app/scripts/modules/core/src/managed/utils/sortEnvironments.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  return 0;
+};
+
+export const sortEnvironments = (response: IManagedApplicationEnvironmentSummary) => {
+  // Sort environment by depends-on constraint. Environments with that constraint should appear first.
+  try {
+    const allEnvironments: { [env: string]: IManagedArtifactVersionEnvironment } = {};
+
+    response.artifacts.map((artifact) => {
+      artifact.versions.map((version) => {
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/managed/utils/sortEnvironments.ts:39]:
```suggestion
    response.artifacts.forEach((artifact) => {
      artifact.versions.forEach((version) => {
```

File: app/scripts/modules/core/src/managed/utils/sortEnvironments.spec.ts (hunk 1 above)
Comment (christopherthielen) [inline app/scripts/modules/core/src/managed/utils/sortEnvironments.spec.ts:81]:
```suggestion
describe('sort environments', () => {
```

File: app/scripts/modules/core/src/managed/utils/sortEnvironments.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    return -1;
+  } else if (isEnvDependsOn(b, a)) {
+    return 1;
+  }
+  // Prioritize if has any depends on
+  const aHasDependsOn = envHasAnyDependsOn(a);
+  const BHasDependsOn = envHasAnyDependsOn(b);
+  if (aHasDependsOn && BHasDependsOn) return 0;
+  if (aHasDependsOn) return -1;
+  if (BHasDependsOn) return 1;
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/managed/utils/sortEnvironments.ts:29]:
```suggestion
  const bHasDependsOn = envHasAnyDependsOn(b);
  if (aHasDependsOn && bHasDependsOn) return 0;
  if (aHasDependsOn) return -1;
  if (bHasDependsOn) return 1;
```
heh

**Review by christopherthielen — APPROVED** [review]
few minor things but otherwise lgtm

File: app/scripts/modules/core/src/managed/utils/sortEnvironments.spec.ts (hunk 1 above)
Comment (ranihorev) [reply app/scripts/modules/core/src/managed/utils/sortEnvironments.spec.ts:81]: I wish there were a way to run a single test via the command line :(

**Pull request merged**